lib/langchain.rb

```diff
@@ -92,24 +91,54 @@
 # Langchain.logger.level = :info
 module Langchain
   class << self
-    # @return [ContextualLogger]
-    attr_reader :logger
-
-    # @param logger [Logger]
-    # @return [ContextualLogger]
-    def logger=(logger)
-      @logger = ContextualLogger.new(logger)
-    end
-
+    # @return [Logger]
+    attr_accessor :logger
     # @return [Pathname]
     attr_reader :root
   end
 
-  self.logger ||= ::Logger.new($stdout, level: :debug)
-
-  @root = Pathname.new(__dir__)
-
   module Errors
     class BaseError < StandardError; end
   end
+
+  module Colorizer
+    class << self
+      def red(str)
+        "\e[31m#{str}\e[0m"
+      end
```

Comment:
Consider maybe converting these methods to the following:
```ruby
COLORS = {
  red: 31,
  green: 32,
  yellow: 33,
  blue: 34
}

def wrap(color, str)
  "\e[#{COLORS[color]}m#{str}\e[0m"
end
```

Reply:
But this is your call!

Reply:
I tried out your suggestion, and I had to change other parts too. In the end the code was a bit less legible and clear. So I'm leaving as it is.